clippy_lints/src/matches/match_same_arms.rs

```diff
@@ -102,45 +103,47 @@ pub(super) fn check<'tcx>(cx: &LateContext<'tcx>, arms: &'tcx [Arm<'_>]) {
 
     let indexed_arms: Vec<(usize, &Arm<'_>)> = arms.iter().enumerate().collect();
     for (&(i, arm1), &(j, arm2)) in search_same(&indexed_arms, hash, eq) {
-        if matches!(arm2.pat.kind, PatKind::Wild) {
-            span_lint_and_then(
-                cx,
-                MATCH_SAME_ARMS,
-                arm1.span,
-                "this match arm has an identical body to the `_` wildcard arm",
-                |diag| {
-                    diag.span_suggestion(arm1.span, "try removing the arm", "", Applicability::MaybeIncorrect)
-                        .help("or try changing either arm body")
-                        .span_note(arm2.span, "`_` wildcard arm here");
-                },
-            );
-        } else {
-            let back_block = backwards_blocking_idxs[j];
-            let (keep_arm, move_arm) = if back_block < i || (back_block == 0 && forwards_blocking_idxs[i] <= j) {
-                (arm1, arm2)
+        if is_lint_allowed(cx, NON_EXHAUSTIVE_OMITTED_PATTERNS, arm2.hir_id) {
```

Comment:
The `non_exhaustive_omitted_patterns` lint is still unstable, which means that this check will fail in almost every project unless it has turned on the specific feature.

You can check if the feature is enabled with `cx.tcx.features().non_exhaustive_omitted_patterns_lint`. In this case, I think it would be good to also add a test file, without the feature, to make sure it behaves like normal in those cases.

I think we should also add a note about this, to the lint documentation. Otherwise, it could get confusing why the lint doesn't trigger in some cases

Reply:
Fixed